server/dash/dashboards/overview_combined.py

```diff
@@ -10,140 +10,240 @@
+        hole=.3,
+        title="Total Requests by Agency",
+    )
+    req_share_by_agency_pie_chart.update_layout(margin=dict(l=100, r=100, b=100, t=100))
+    return req_share_by_agency_pie_chart
 
 # Request Type by Source Bar Chart.
-req_source_count_df = req_source_count
-req_source_count_df.sort_values('counts', ascending=False, inplace=True)
-req_source_count_df.loc['Others'] = req_source_count_df[NON_TOP4_INDEX_START:].sum()
-req_source_count_df.sort_values('counts', ascending=False, inplace=True)
-req_source_count_df = req_source_count_df[:NON_TOP4_INDEX_START + 1]
-
-req_source_bar_chart = px.bar(
-    req_source_count_df,
-    y=req_source_count_df.index,
-    x='counts',
-    title="Total Requests by Source",
-    orientation='h'
-)
+def make_req_type_source_bar_chart(req_source_count):
+    """Generates the request type by source bar chart.
+
+    This function takes the "req_source_count" dataframe, aggregate all request source with less than 
+    top 4 request counts as "Others", and outputs a horizontal bar chart "req_source_bar_chart" showing
+    the share of requests by sources.
+
+    Args:
+        req_source_count: a pandas dataframe with the different request sources and corresopnding request counts.
+
+    Returns: 
+        req_source_bar_chart: a horizontal bar chart showing the share of requests by sources.
+    """
+    req_source_count_df = req_source_count
+    req_source_count_df.sort_values('counts', ascending=False, inplace=True)
+    req_source_count_df.loc['Others'] = req_source_count_df[NON_TOP4_INDEX_START:].sum()
+    req_source_count_df.sort_values('counts', ascending=False, inplace=True)
+    req_source_count_df = req_source_count_df[:NON_TOP4_INDEX_START + 1]
+
+    req_source_bar_chart = px.bar(
+        req_source_count_df,
+        y=req_source_count_df.index,
+        x='counts',
+        title="Total Requests by Source",
+        orientation='h'
+    )
+    return req_source_bar_chart
 
 # Median Request Days to Close Box Plot.
-print(" * Downloading data from API path: " + REQ_TYPE_STATS_API_PATH)
-stats_df = pd.read_json(API_HOST + REQ_TYPE_STATS_API_PATH)
-stats_df = stats_df.sort_values('median', ascending=False)
-print(" * Dataframe has been loaded from API path: " + REQ_TYPE_STATS_API_PATH)
-
-med_days_to_close_box_plot = go.Figure()
-med_days_to_close_box_plot.add_trace(
-    go.Box(
-        y=stats_df.type_name,
-        q1=stats_df['q1'],
-        median=stats_df['median'],
-        q3=stats_df['q3'],
-        marker_color='#29404F',
-        fillcolor='#E17C05',
+def make_days_to_close_box_plot():
+    """Generates the request days to close box plot.
+
+    This function calls the stats api via "REQ_TYPE_STATS_API_PATH", retrieves quartile values for request
+    time to close, and generates a box plot to visualize median request day to close box plot.
+
+    Returns: 
+        med_days_to_close_box_plot: a box plot showing the median day to close for each request type.
```

Comment:
nit: In the "Returns" section of a docstring, if we are only returning one thing, we don't need to name it.

Reply:
Thanks for the suggestion - resolved.